From 9947d30b849f1674775798b597e7a235fb754a21 Mon Sep 17 00:00:00 2001
From: Benoit Fradin <benoit.fradin@intel.com>
Date: Wed, 28 Aug 2013 14:44:32 +0200
Subject: [PORT FROM MAIN] Fix KW critical errors in canonicalize_slashes

BZ: 134244

Add a return case for len == 0 and obsolete == false

Change-Id: Ia36bd67ced6abbe7234eb7e485c1a2d8bd928ffe
Orig-Change-Id: I18e50cc87b8a6809c339b4df8ddc5ba34396bcdb
Category: aosp improvement
Domain: PUPDR-Recovery
Origin: internal
Upstream-Candidate: no, need rework
Signed-off-by: Benoit Fradin <benoit.fradin@intel.com>
Reviewed-on: http://android.intel.com:8080/128274
Reviewed-by: Touzeau, GabrielX <gabrielx.touzeau@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Lumbierres, PhilippeX <philippex.lumbierres@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 ext4_utils/make_ext4fs.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/ext4_utils/make_ext4fs.c b/ext4_utils/make_ext4fs.c
index cb7424e..c031de4 100644
--- a/ext4_utils/make_ext4fs.c
+++ b/ext4_utils/make_ext4fs.c
@@ -409,8 +409,11 @@ static char *canonicalize_slashes(const char *str, bool absolute)
 	int newlen = len;
 	char *ptr;
 
-	if (len == 0 && absolute) {
-		return strdup("/");
+	if (len == 0) {
+		if (absolute)
+			return strdup("/");
+		else
+			return strdup("");
 	}
 
 	if (str[0] != '/' && absolute) {
-- 
1.7.4.1

